applications/ChimeraApplication/custom_utilities/distance_calcuation_utility.h

```diff
@@ -0,0 +1,176 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Aditya Ghantasala
+
+#if !defined(DISTANCE_CALCULATION_UTILITY )
+#define  DISTANCE_CALCULATION_UTILITY
+
+
+
+// System includes
+
+
+// External includes
+#include "omp.h"
+
+
+// Project includes
+#include "includes/define.h"
+// #include "factories/standard_linear_solver_factory.h"
+#include "processes/variational_distance_calculation_process.h"
+#include "utilities/parallel_levelset_distance_calculator.h"
+#include "processes/calculate_distance_to_skin_process.h"
+#include "utilities/variable_utils.h"
+
+
+namespace Kratos
+{
+
+///@name Kratos Classes
+///@{
+
+/// Utility for calculating the Distance on a given modelpart
+template <int TDim>
+class KRATOS_API(CHIMERA_APPLICATION) DistanceCalculationUtility
```

Comment:
```suggestion
class KRATOS_API(CHIMERA_APPLICATION) ChimeraDistanceCalculationUtility
```
To avoid misunderstandings with the ones in the Kratos Core.

Again I'm wondering why do we need a specific distance calculation for the Chimera. Can you clarify please?

Reply:
OK. Now I see. Aside of calculating the level set, you also want to smooth it by using the Parallel distance calculation.

1. I think it would be better to call both utilities one after the other rather than creating a specific class for this.
2. If you want to spread the level set, maybe the Variational distance can do a better job. @RiccardoRossi

Reply:
name change makes sense ! Done. 
Regarding the why a separate file is needed. we can discuss in person !